Card/src/main/java/com/paypal/android/card/CardRequestFactory.kt

```diff
@@ -30,18 +30,25 @@ internal class CardRequestFactory {
                 cardJSON.put("billing_address", billingAddressJSON)
             }
 
+            val bodyJSON = JSONObject()
             threeDSecureRequest?.also {
                 val verificationJSON = JSONObject()
                     .put("method", it.sca.name)
                 val attributesJSON = JSONObject()
                     .put("verification", verificationJSON)
                 cardJSON.put("attributes", attributesJSON)
+
                 // add return and cancel url when its supported
+                val returnURLJSON = JSONObject()
+                    .put("return_url", it.returnUrl)
+                    .put("cancel_url", it.cancelUrl)
+                bodyJSON.put("application_context", returnURLJSON)
```

Comment:
We should update our `CardRequestFactoryUnitTests` to make sure these URLs are properly set in the request body to `confirm-payment-source`.

Reply:
Where do `returnUrl` and `cancelUrl` come from? It seems like we used to expect the merchant to set them in the Fragment, but now we're getting them elsewhere?

Reply:
Yeah i'm not sure why it isn't showing in the diff I guess we had some duplicated logic. We still do still [set them in CardFragment](https://github.com/paypal/Android-SDK/blob/8ae947a7b8f01f92e104e47971a1cb1f9135b188/Demo/src/main/java/com/paypal/android/ui/card/CardFragment.kt#L201).

Reply:
And the "when it's supported" is referencing the bug. That should be a TODO I can remove that now.

Reply:
> We should update our `CardRequestFactoryUnitTests` to make sure these URLs are properly set in the request body to `confirm-payment-source`.

Updated the `CardRequestFactoryUnitTests` to reflect the new change.